src/openms_gui/source/VISUAL/Spectrum1DCanvas.cpp

```diff
@@ -1130,7 +1130,8 @@ namespace OpenMS
     {
       (*getCurrentLayer_().getPeakDataMuteable())[i].sortByPosition(); // TODO 
     }
-    getCurrentLayer_().getCurrentSpectrumMuteable().sortByPosition();
+    Size current_spectrum = getCurrentLayer_().getCurrentSpectrumIndex();
+    getCurrentLayer_().getPeakDataMuteable()->getSpectrum(current_spectrum).sortByPosition();
```

Comment:
this is actually now not the same thing as before, as we were sorting the cached spectrum